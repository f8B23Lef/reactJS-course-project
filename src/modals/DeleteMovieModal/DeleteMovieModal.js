import React from 'react';
import { useDispatch } from 'react-redux';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faTimes } from '@fortawesome/free-solid-svg-icons';
import { deleteMovie } from '../../store/moviesSlice';
import './DeleteMovieModal.scss';

export default function DeleteMovieModal(props) {
  const {
    movieId,
    onClose,
  } = props;
  const dispatch = useDispatch();

  const onConfirm = () => {
    dispatch(deleteMovie(movieId));
    onClose();
  };

  return (
    <>
      <div className='modal-header'>
        <h3 className='modal-title'>Delete Movie</h3>
        <FontAwesomeIcon
          icon={faTimes}
          className='close__icon'
          data-testid='close-icon'
          onClick={onClose}
        />
      </div>
      <div className='modal-body'>
        <div className='message'>
          Are you sure you want to delete this movie?
        </div>
      </div>
      <div className='modal-footer'>
        <button
          type='button'
          className='submit__button'
          onClick={onConfirm}
        >
          Confirm
        </button>
      </div>
    </>
  );
}

DeleteMovieModal.propTypes = {
  movieId: PropTypes.number.isRequired,
  onClose: PropTypes.func.isRequired,
};
