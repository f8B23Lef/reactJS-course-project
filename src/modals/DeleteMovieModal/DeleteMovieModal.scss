@import '../../styles/colors.scss';
@import '../../styles/modals.scss';

.modal-body .message {
  margin: 10px;
  color: $white;
}
