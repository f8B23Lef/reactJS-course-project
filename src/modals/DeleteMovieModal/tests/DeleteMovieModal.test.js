import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render, screen, fireEvent } from '@testing-library/react';
import DeleteMovieModal from '../DeleteMovieModal';

const mockedUseDispatch = jest.fn();
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useDispatch: () => mockedUseDispatch,
}));

describe('DeleteMovieModal', () => {
  const onCloseMock = jest.fn();
  const movieId = 1;

  it('renders DeleteMovieModal snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <DeleteMovieModal
          movieId={movieId}
          onClose={onCloseMock}
        />
      </BrowserRouter>,
    );

    expect(asFragment(<DeleteMovieModal
      movieId={movieId}
      onClose={onCloseMock}
    />)).toMatchSnapshot();
  });

  it('calls onClose() on close icon click', () => {
    render(
      <BrowserRouter>
        <DeleteMovieModal
          movieId={movieId}
          onClose={onCloseMock}
        />
      </BrowserRouter>,
    );

    const closeIcon = screen.getByTestId('close-icon');
    fireEvent.click(closeIcon);

    expect(onCloseMock).toHaveBeenCalled();
  });

  it('calls dispatch() on Confirm button click', () => {
    render(
      <BrowserRouter>
        <DeleteMovieModal
          movieId={movieId}
          onClose={onCloseMock}
        />
      </BrowserRouter>,
    );

    const confirmButton = screen.getByRole('button');
    fireEvent.click(confirmButton);

    expect(mockedUseDispatch).toHaveBeenCalled();
  });

  it('calls onClose() on Confirm button click', () => {
    render(
      <BrowserRouter>
        <DeleteMovieModal
          movieId={movieId}
          onClose={onCloseMock}
        />
      </BrowserRouter>,
    );

    const confirmButton = screen.getByRole('button');
    fireEvent.click(confirmButton);

    expect(onCloseMock).toHaveBeenCalled();
  });
});
