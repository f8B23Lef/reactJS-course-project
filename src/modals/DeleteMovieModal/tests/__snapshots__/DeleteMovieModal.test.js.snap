// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteMovieModal renders DeleteMovieModal snapshot 1`] = `
<DocumentFragment>
  <div
    class="modal-header"
  >
    <h3
      class="modal-title"
    >
      Delete Movie
    </h3>
    <svg
      aria-hidden="true"
      class="svg-inline--fa fa-xmark close__icon"
      data-icon="xmark"
      data-prefix="fas"
      data-testid="close-icon"
      focusable="false"
      role="img"
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M310.6 150.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L160 210.7 54.6 105.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3L114.7 256 9.4 361.4c-12.5 12.5-12.5 32.8 0 45.3s32.8 12.5 45.3 0L160 301.3 265.4 406.6c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L205.3 256 310.6 150.6z"
        fill="currentColor"
      />
    </svg>
  </div>
  <div
    class="modal-body"
  >
    <div
      class="message"
    >
      Are you sure you want to delete this movie?
    </div>
  </div>
  <div
    class="modal-footer"
  >
    <button
      class="submit__button"
      type="button"
    >
      Confirm
    </button>
  </div>
</DocumentFragment>
`;
